DataProc/Common/cloneraw.C

```diff
@@ -0,0 +1,79 @@
+void cloneraw(const char* fileName, const char* chunkName) {
+
+  // --- "Analysis part" ---
+  // This has to be done in the analysis task that produces the TEntryList
+
+  // from env variables
+  Int_t runNumber = atoi(gSystem->Getenv("ALIEN_JDL_LPMRUNNUMBER"));
+  Int_t year = atoi(gSystem->Getenv("ALIEN_JDL_LPMANCHORYEAR"));
+  TString period = gSystem->Getenv("ALIEN_JDL_LPMPRODUCTIONTAG");
+  TString pass = gSystem->Getenv("ALIEN_JDL_LPMPASSNAME");
+  
+  // ESD file and tree
+  TGrid::Connect("alien://");
+  TString filename(TString::Format("alien:///alice/data/%d/%s/%09d/%s/%s/AliESDs.root", year, period.Data(), runNumber, pass.Data(), chunkName));
+  cout << "Opening ESD file " << filename << endl;
+  TFile* fesd = TFile::Open(TString::Format("%s", filename.Data()));
+  //  TFile fesd("AliESDs.root");
+  TTree * tesd = (TTree*)fesd->Get("esdTree");
+
+  // ESD event
+  AliESDEvent * esd = new AliESDEvent();// The signal ESD object is put here
+  esd->ReadFromTree(tesd);
+
+  // TEntryList
+  TEntryList fLocalList;
+  fLocalList.SetTreeName("RAW");
+  fLocalList.SetFileName(fileName);
+
+  // reading file with mapping
+  ifstream inFile;
+  inFile.open(TString::Format("ListOfFiles_%s.txt", period.Data()));
+  if (!inFile) {
+    cerr << "Unable to open file for mapping";
+    exit(1);   // call system to stop
+  }
+  string line;
+  bool found = false;
+  while (getline(inFile, line)) {
+    //cout << "line = " << line << endl;
+    if (line.find(chunkName) != std::string::npos) {
+      TString strTmp(line);
+      TObjArray* arr = strTmp.Tokenize(":");
+      TObjString* objstr = (TObjString*)arr->At(2);
+      //cout << "token 2: " << objstr->String() << endl;
+      Int_t nevToFilter = (objstr->String()).Atoi();
+      fLocalList.Enter(nevToFilter);
+      cout << "Found event " << nevToFilter << " for chunk " << chunkName << endl; 
+      found = true;
+      delete arr;
+    }
+    else if (found == true) break;

```

Comment:
Isn't this condition preventing from finding the 2nd line/event in those cases in which there are two events from the same chunk?

Reply:
Ciao, 

The break is called when we don't find anymore the wanted chunk in the line, so when there are multiple events, it is not yet called, unless the chunks are not in order. @mpuccio , @akalweit, can you confirm that when a chunk has more than one event, they are all consecutives? In the file, the order is not alphabetical. We can also change the file, and make it alphabetical, to be sure. Or we loop over all lines. 

Let me know what you prefer.

Chiara

Reply:
The new commit takes as input files that are sorted alphabetically - there is no real change in the logic, just the file name. So, beforehand, we should sort the file. 
But as I said, if we want to be 100% sure, we can go through all the lines every time.

Reply:
Hi,
I do not have a strong opinion on that. For me, it would also be fine to sort the file beforehand.

Thanks a lot,
Alexander